workers/unity/Assets/Playground/Scripts/Bootstrap.cs

```diff
@@ -136,13 +126,18 @@ public static void SetupInjectionHooks()
 
         public static void DomainUnloadShutdown()
         {
+            // because otherwise the world won't get cleaned up and unity stops working:
+            foreach (var world in WorldToConfig.Keys) {
+                world.Dispose();
+            }
+
             World.DisposeAllWorlds();
```

Comment:
I think `World.DisposeAllWorlds` has a similar effect to above.

Now that we do the specific world disposal it will be nice to get rid of `World.DisposeAllWorlds()` as well - as it has other side effects of cleaning user-generated worlds too.